pytoda/smiles/transforms.py

```diff
@@ -3,24 +3,164 @@
+                Kekulize(
+                    all_bonds_explicit=all_bonds_explicit,
+                    all_hs_explicit=all_hs_explicit,
+                    sanitize=sanitize
+                )
+            ]
+        elif all_bonds_explicit or all_hs_explicit:
+            smiles_transforms += [
+                NotKekulize(
+                    all_bonds_explicit=all_bonds_explicit,
+                    all_hs_explicit=all_hs_explicit,
+                    sanitize=sanitize
+                )
+            ]
+        if augment:
+            smiles_transforms += [
+                Augment(
+                    kekule_smiles=kekulize,
+                    all_bonds_explicit=all_bonds_explicit,
+                    all_hs_explicit=all_hs_explicit,
+                    sanitize=sanitize
+                )
+            ]
+        if selfies:
+            smiles_transforms += [Selfies()]
+
+    return Compose(smiles_transforms)
+
+
+def compose_encoding_transforms(
+    randomize: bool = False,
+    add_start_and_stop: bool = False,
+    start_index: int = 2,
+    stop_index: int = 3,
+    padding: bool = False,
+    padding_length: int = None,
+    padding_index: int = 0,
+    device: torch.device = torch.
+        device('cuda' if torch.cuda.is_available() else 'cpu'),
+) -> Tuple[Compose, Callable]:
+    """Setup a composition of token indices to token indices transformations.
+
+    Args:
+        randomize (bool, optional): perform a true randomization of
+            token indexes. Defaults to False.
+        add_start_and_stop (bool, optional): add start and stop token
+            indexes. Defaults to False.
+        start_index (int, optional): index of start token in vocabulary.
+            Default to 2.
+        stop_index (int, optional): index of stop token in vocabulary.
+            Default to 3.
+        padding (bool, optional): pad sequences to given padding_length.
+            Defaults to True.
+        padding_length (int, optional): manually sets number of applied
+            paddings, applies only if padding is True. Defaults to None, but
+            must be passed in case of padding.
+        padding_index (int, optional): index of padding token in vocabulary.
+            Default to 0.
+        device (torch.device): device where the tensors are stored.
+            Defaults to gpu, if available.
+
+    Returns:
+        Compose: A Callable that applies composition of transforms on
+            token indices.
+
+    Note:
+        Transformations can change the number of tokens.
+    """
+    encoding_transforms = []
+
+    if randomize:
+        encoding_transforms += [Randomize()]
+    if add_start_and_stop:
+        encoding_transforms += [StartStop(start_index, stop_index)]
+
+    if padding:
+        encoding_transforms += [
+            LeftPadding(
+                padding_length=padding_length,
+                padding_index=padding_index
+            )
+        ]
+    # TODO conditional?
```

Comment:
is this a todo?

Reply:
A TODO as in I wanted to discuss this.
I was thinking that the DataLoaders default collate_fn is taking care of returning tensors from many types
>tensors, numpy arrays, numbers, dicts or lists
So the tokenizers / datasets don't necessarily have to. But then is it on the correct device?
We should discuss briefly in person.